frontend/frontend/controllers/scanctrl.py

```diff
@@ -40,6 +40,48 @@
 # ===================
 
 
+def get_tagid(session, text):
+    tag_list = Tag.query_find_all(session)
+    
+    for tag in tag_list:
+        if tag.text == text:
+            log.debug("get_tagid :: tag [%s] found with id %d", text, id)
+            return tag.id
+
+    log.debug("get_tagid :: tag [%s] not found!", text)
+    return 0
+
+
+
+def set_probe_tag(file_hash, probe, result):
+
+
+    # If the probe is an antivirus and the result value is "1" (infected)
+    if result['type'] == 'antivirus' and result['status'] == 1 :
+
+        with session_transaction() as session:
+            available_tags = Tag.query_find_all(session)
+            if probe in [t.text for t in available_tags]:
+                log.debug("set_probe_tag :: tag [%s] already exists", probe)
+            else:
+                # create tag for the probe.
+                log.debug("set_probe_tag :: adding tag [%s] to database", probe)
+                tag = Tag(probe)
+                session.add(tag)
+                session.commit()
+
+            # add tag to file.
+            log.debug("set_probe_tag :: tagging file [%s] with tag [%s]",file_hash, probe)
+            file = File.load_from_sha256(file_hash, session)
+
+            tag_id = get_tagid(session, probe)
+            if tag_id != 0 :
```

Comment:
If you're using the `try: … except` method above, then `tag` variable (and `tag.id`) can't be empty (it will depend on the code you've put in the `except` blocks), so you can add directly the tag.